routing/router.go

```diff
@@ -1138,61 +1143,32 @@ func (r *ChannelRouter) networkHandler() {
 
 			// We'll ensure that any new blocks received attach
 			// directly to the end of our main chain. If not, then
-			// we've somehow missed some blocks. We don't process
-			// this block as otherwise, we may miss on-chain
-			// events.
+			// we've somehow missed some blocks. Here we'll catch
+			// up the chain with the latest blocks.
 			currentHeight := atomic.LoadUint32(&r.bestHeight)
-			if chainUpdate.Height != currentHeight+1 {
+			if chainUpdate.Height == currentHeight+1 {
+				err := r.updateGraphWithClosedChannels(
+					chainUpdate,
+				)
+				if err != nil {
+					log.Errorf("unable to prune graph "+
+						"with closed channels: %v", err)
+				}
+			} else {
```

Comment:
This would nice and readable if we just had one switch with all the different height cases: 
```
switch {
  case chainUpdate.Height == currentHeight+1: 
       updateGraphWithClosedChannels ...
       
  case chainUpdate.Height > currentHeight+1:
  
  etc
```